docs/control-structures.rst

```diff
@@ -101,18 +101,18 @@ The ``@nonreentrant(<key>)`` decorator places a lock on a function, and all func
         # this function is protected from re-entrancy
         ...
 
-You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function but we recommend against it because in most circumstances it will not work in a meaningful way.
+You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function, but we recommend against it because in most circumstances it will not work as intended.
```

Comment:
not sure why this change was made?

Reply:
Actually, even the correction is wrong. It should be the following instead:

```diff
+You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function; but we recommend against it, because in most circumstances it will not work as intended.
```
Rationality: 

- `You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function;` is an independent clause, and leads to a comma splice error. To avoid this, we add a `;`
- `because in most circumstances it will not work as intended.` is a relative clause which adds to the first part (or `but we recommend against it,` so we add a comma here).
- As for the change to `as intended` -- it is so for concision.